simpleflow/swf/process/worker/base.py

```diff
@@ -331,7 +341,13 @@ def spawn(poller, token, task, middlewares=None, heartbeat=60):
     :param heartbeat: heartbeat delay (seconds)
     :type heartbeat: int
     """
-    logger.info("spawning new activity worker pid={} heartbeat={}".format(os.getpid(), heartbeat))
+    logger.info(
+        "spawning new activity id=%s worker pid=%d heartbeat=%s",
+        task.activity_id,
+        os.getpid(),
```

Comment:
I believe displaying the pid should be the role of the log formatter, not the log message itself. https://docs.python.org/3/library/logging.html#logrecord-attributes

Reply:
... and it is. Good catch!

Reply:
Not sure how to handle this though, as we integrate Simpleflow in a Django project, and inherit from the logger configured in the project's `settings` :thinking:

Reply:
Maybe add `"formatter": "simpleflow_formatter",` to the Django project's settings? What could go wrong? 🙃

Reply:
Double-checked:

- we do not override Simpleflow's logging options in our workers
- `logging` is initialized with `setup_logging()` at https://github.com/botify-labs/simpleflow/blob/main/simpleflow/log.py#L148 , that relies on the default `LOGGING` dict at https://github.com/botify-labs/simpleflow/blob/main/simpleflow/settings/default.py#L27
- this sets up `SimpleflowFormatter` https://github.com/botify-labs/simpleflow/blob/main/simpleflow/log.py#L49 as the default formatter
- workers use `loggers` obtained from `logging.getLogger()`

What I'm not clear on though, is whether we use:

- the format string set in https://github.com/botify-labs/simpleflow/blob/main/simpleflow/settings/default.py#L50
- the format string set in https://github.com/botify-labs/simpleflow/blob/main/simpleflow/log.py#L87

Reply:
Looks Logged To Me :+1: